third_party/nvfuser/csrc/fusion_segmenter.h

```diff
@@ -124,6 +132,7 @@ class TORCH_CUDA_CU_API SegmentedGroup {
   friend class SegmentedFusion;
   friend class FusionKernelRuntime;
   friend class TranslateApplicableWelford;
+  friend class MultiGroupFusion;
```

Comment:
Wonder if fusion segmenter needs to be more drastically refactored.

Reply:
It would be nice if there was a consistent way to reuse the segmentation infrastructure for this work.

Reply:
an introduction on how this segmenter works would be great !